luncheon/headers.lua

```diff
@@ -0,0 +1,173 @@
+---@class Headers
+---
+---A map of the key value pairs from the header portion
+---of an HTTP request or response. The fields listed below
+---are some common headers but the list is not exhaustive.
+---
+---When each header is serialized, it is added as a property with
+---lower_snake_case. For example, `X-Forwarded-For` becomes `x_forwarded_for`.
+---
+---Typically each header's value will be a string, however if there are multiple entries
+---for any header, it will be a table of strings.
+---
+---@field private _inner table A table containing the deserialized header key/value pairs
+---@field private _last_key string The last key deserialized, for multiline headers
+local Headers = {}
+
+Headers.__index = Headers
+
+local function _append(t, key, value)
+    value = tostring(value)
+    if not t[key] then
+        t[key] = value
+    elseif type(t[key]) == 'string' then
+        t[key] = {t[key], value}
+    else
+        table.insert(t[key], value)
+    end
+end
+
+---Serialize a key value pair
+---@param key string
+---@param value string
+---@return string
+function Headers.serialize_header(key, value)
+    if type(value) == 'table' then
+        value = value[#value]
```

Comment:
If there is more than one value for a header, only the last one added will be serialized. I think all the values can be included as a comma separated list. I faced this small issue while working with lustre. I would be fine leaving as is for now, but at some point we probably want to serialize multiple values correctly.

Reply:
Good catch. I'll add an issue for this